### DISCUSSION TITLE
Fetch list of players?

### DISCUSSION THREAD

**Discussion by pritish-devurkar** [cwendt94/espn-api#261] [Q&A]
How would I go about retrieving the entire list of available players?

**Comment by pritish-devurkar** [comment]
Found the answer, the league object has a player_map attribute, which is a dictionary of all playerIds and names

**Reply by pritish-devurkar** [reply]
There seems to be some disparity in the data returned by the player_map:

3915487, 'Mike Danna', 4423367, **'Anthony Pittman',** **16140, 16143, 'Josh Hill'**, 4046652, 'Chris Wilcox', 3915575, 'Kelvin McKnight', 16166,

**Reply by pritish-devurkar** [reply]
is there a better way to retrieve the entire player list?

**Reply by cwendt94** [reply]
Hmm I wonder why. You can get the raw data for all active players with this call 
```python3
league.espn_request.get_pro_players()
```

**Reply by pritish-devurkar** [reply]
I ended up using something like this to get all rostered players:

```
allrosteredplayers = []
    for team in league.teams:
        for player in team.roster:
            allrosteredplayers.append(player)
```

**Reply by cwendt94** [reply]
That will definitely work and you can also get all of the free agents as well.
``` python3
league.free_agents()
```

**Reply by pritish-devurkar** [reply]
Sure, thanks!
Also btw thank you so much for creating and maintaining this library 👍 

I'm trying to build a trade analyzer tool based on top of this

**Reply by cwendt94** [reply]
No problem! I have been pretty busy lately so I have not been able to add more features as I would have liked. 

I am planning on doing a large refactor in the next couple months that will combine a lot of common fields and features across all of the sports and make it easier to update.

That sounds awesome, looking forward to hearing about the finished product!
